src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useContext, useMemo } from 'react';
+import { TodoList } from '../TodoList/TodoList';
+import { Header } from '../Header/Header';
+import { Footer } from '../Footer/Footer';
+import { TodosContext } from '../../contextes/TodosContext';
+
+export const TodoApp = () => {
+  const { todos, setTodos, filterField } = useContext(TodosContext);
+
+  const conuterOfCompletedTodos = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length === 0;
+  }, [todos]);
+
+  const toggledTodosList = useMemo(() => {
+    if (conuterOfCompletedTodos) {
+      return todos.map(todo => ({
+        ...todo,
+        completed: false,
+      }));
+    }
+
+    return todos.map(todo => ({
+      ...todo,
+      completed: true,
+    }));
+  }, [todos, conuterOfCompletedTodos]);
+
+  const preparedTodosList = todos.filter(todo => {
+    switch (filterField) {
+      case 'Active':
+        return todo.completed === false;
+
+      case 'Completed':
+        return todo.completed === true;
+
+      default:
+        return todo;
+    }
+  });
+
+  return (
+    <div className="todoapp">
+      <Header />
+
+      <section className="main">
+        {!todos.length || (
```

Comment:
```suggestion
        {!todos.length && (
```
Why || ?

Reply:
You use **!** in this case.  
That means you have a boolean condition that can`t render in jsx.